go/vt/vtgate/executor.go

```diff
@@ -1140,7 +1157,9 @@ func (e *Executor) VSchema() *vindexes.VSchema {
 func (e *Executor) SaveVSchema(vschema *vindexes.VSchema, stats *VSchemaStats) {
 	e.mu.Lock()
 	defer e.mu.Unlock()
-	e.vschema = vschema
+	if vschema != nil {
+		e.vschema = vschema
+	}
```

Comment:
This can create an issue when a user wants to explicitly clear out the VSchema.
This check will ignore the update and will keep the old version.